test/trans_interop.py

```diff
@@ -136,21 +261,32 @@ def run_test(self):
         print("mempool counts: %s" % [ x.getmempoolinfo()["size"] for x in self.nodes])
 
         print("Verify transaction amounts and confirmation counts between two nodes")
-        # Simple send amount = 10 from Node 0 to 1 
-        verify_transaction_amount_confirm(self, 0, 1, 10)
+        verify_sendto_different_node(self, 10)
+        
+        print("Verify transaction amounts on the same nodes")
+        verify_sendto_same_node(self, 10)
         
-        # Simple send amount = 10 from Node 0 to 2 
-        verify_transaction_amount_confirm(self, 0, 2, 10)
+        print("Verify transaction amounts and confirmation counts between two nodes")
+        verify_sendto_different_node(self, 0.5)
         
-        # Simple send amount = 10 from Node 1 to 2 
-        verify_transaction_amount_confirm(self, 1, 2, 10)
+        print("Verify transaction amounts on the same nodes")
+        verify_sendto_same_node(self, 0.5)
         
-        # Simple send amount = 10 from Node 1 to 3 
-        verify_transaction_amount_confirm(self, 1, 3, 10)
+        time.sleep(10)
```

Comment:
why is a sleep needed here?  sleeping is dangerous because different systems need different sleep times.  Its better to explicitly check for whatever condition you are waiting for.

Reply:
Yes, this sleep is not needed. I forgot to remove it.
Retested ok with calling verify_transaction_sendmany(self, 0, 1, 1) just once within this test case.

But when I call this function 3 times with different arguments as shown, 
verify_transaction_sendmany(self, 0, 1, 1)
verify_transaction_sendmany(self, 1, 2, 1)
verify_transaction_sendmany(self, 2, 3, 1)

I keep getting JSONRPC error: importaddress "address" ( "label" rescan ) as shown below.
The reason I added the sleep(10) is because the log shows "this call can take minutes to complete if rescan is true."  It seems to help, but still fail the same way sometimes.

I think I should be able to call this function "verify_transaction_sendmany" many times and with different input combo. 

Have you seen this problem before? Is there anything I have missed or I am not testing it correctly.

Thanks


------------------ error log ---------------------------------------------------------------
JSONRPC error: importaddress "address" ( "label" rescan )

Adds an address or script (in hex) that can be watched as if it were in your wallet but cannot be used to spend.

Arguments:
1. "address"          (string, required) The address
2. "label"            (string, optional, default="") An optional label
3. rescan               (boolean, optional, default=true) Rescan the wallet for transactions

Note: This call can take minutes to complete if rescan is true.

Examples:

Import an address with rescan
> bitcoin-cli importaddress "myaddress"

Import using a label without rescan
> bitcoin-cli importaddress "myaddress" "testing" false

As a JSON-RPC call
> curl --user myusername --data-binary '{"jsonrpc": "1.0", "id":"curltest", "method": "importaddress", "params": ["myaddress", "testing", false] }' -H 'content-type: text/plain;' http://127.0.0.1:8332/

  File "/home/sk1/Dev/cashInterop/test/test_framework/test_framework.py", line 192, in main
    self.run_test()
  File "./trans_interop.py", line 287, in run_test
    verify_transaction_sendmany(self, 2, 3, 1)
  File "./trans_interop.py", line 75, in verify_transaction_sendmany
    self.nodes[nodeTxOut].importaddress(multisig["redeemScript"], "watchonly", False, True)
  File "/home/sk1/Dev/cashInterop/test/test_framework/coverage.py", line 49, in __call__
    return_val = self.auth_service_proxy_instance.__call__(*args, **kwargs)
  File "/home/sk1/Dev/cashInterop/test/test_framework/authproxy.py", line 165, in __call__
    raise JSONRPCException(response['error'])
> /home/sk1/Dev/cashInterop/test/test_framework/authproxy.py(165)__call__()
-> raise JSONRPCException(response['error'])
(Pdb)

Reply:
clientDirs = ["xt", "xt", "xt", "xt"] 
clientDirs = ["xt", "abc", "bucash", "bucash"]
clientDirs = ["bucash", "abc", "xt", "bucash"] 

The JSONRPC error seems to be caused by XT client 
Whenever XT is used on the list above, I got the same error